lib/ffi-icu/time_formatting.rb

```diff
@@ -0,0 +1,176 @@
+                                          #   then the short localized GMT format as the final fallback., such as "ET".
+            :specific_long    =>  'zzzz', # The long specific non-location format.
+                                          #   Where that is unavailable, falls back to the long localized GMT format ("OOOO").
+            :specific_short   =>     'z', # The short specific non-location format.
+                                          #   Where that is unavailable, falls back to the short localized GMT format ("O").
+            :basic            =>     'Z', # The ISO8601 basic format with hours, minutes and optional seconds fields.
+                                          #   The format is equivalent to RFC 822 zone format (when optional seconds field is absent).
+                                          #   This is equivalent to the "xxxx" specifier.
+            :localized_long   =>  'ZZZZ', # The long localized GMT format. This is equivalent to the "OOOO" specifier, such as GMT-8:00
+            :extended         => 'ZZZZZ', # The ISO8601 extended format with hours, minutes and optional seconds fields.
+                                          #   The ISO8601 UTC indicator "Z" is used when local time offset is 0.
+                                          #   This is equivalent to the "XXXXX" specifier, such as -08:00 -07:52:58
+            :localized_short  =>    'O',  # The short localized GMT format, such as GMT-8
+            :localized_longO  => 'OOOO',  # The long localized GMT format, such as GMT-08:00
+            :tz_id_short      =>    'V',  # The short time zone ID. Where that is unavailable,
+                                          #   the special short time zone ID unk (Unknown Zone) is used.
+                                          #   Note: This specifier was originally used for a variant of the short specific non-location format,
+                                          #   but it was deprecated in the later version of this specification. In CLDR 23, the definition
+                                          #   of the specifier was changed to designate a short time zone ID, such as uslax
+            :tz_id_long       =>   'VV',  # The long time zone ID, such as America/Los_Angeles
+            :city_location    =>  'VVV',  # The exemplar city (location) for the time zone. Where that is unavailable,
+                                          #   the localized exemplar city name for the special zone Etc/Unknown is used as the fallback
+                                          #   (for example, "Unknown City"), such as Los Angeles
+            # see: http://unicode.org/reports/tr35/tr35-dates.html#Date_Format_Patterns
+          }
+          repl = $tz_map[tz_style]
+          raise 'no such tz_style' unless repl
+          tz.gsub!(/^(\s*)(.*?)(\s*)$/, '\1'+repl+'\3')
+        end
+        return pre + tz + suff
+        format
```

Comment:
Line 145 can't be reached and should be removed. 
The explicit `return` keyword is redundant and should be removed.